src/main/java/com/crowdin/client/reports/ReportsApi.java

```diff
@@ -216,4 +216,101 @@ public ResponseObject<ReportSettingsTemplate> editReportSettingsTemplate(Long pr
     public void deleteReportSettingsTemplate(Long projectId, Long reportSettingsTemplateId) throws HttpException, HttpBadRequestException {
         this.httpClient.delete(this.url + "/projects/" + projectId + "/settings-templates/" + reportSettingsTemplateId, new HttpRequestConfig(), Void.class);
     }
+
+    // -- REPORT ARCHIVES -- //
```

Comment:
There is also a group or Organization reports APIs (for Crowdin Enterprise - https://developer.crowdin.com/enterprise/api/v2/#operation/api.reports.archives.getMany)

These methods are almost the same, the only difference is the request URI (it doesn't contain the `users` sub-path)

We should support these methods as well. Probably add the `Organization` to the method name, as like we have in `generateOrganizationReport`, `checkOrganizationReportGenerationStatus`, etc.

Reply:
Or we can try making the `userId` optional and depending on its value build the request URI

Reply:
I think making userId optional is a great idea, since endpoinds are the same and the only diff is 'userId'. By that we can avoid a repetetive code and have one endpoint for both.

` public ResponseList<ReportArchive> getListReportArchives(Long userId, String scopeType, Long scopeId, Integer limit, Integer offset) throws HttpException, HttpBadRequestException {
        String url = userId != null ? this.url + "/users/" + userId : this.url;
        Map<String, Optional<Object>> queryParams = HttpRequestConfig.buildUrlParams(
                "scopeType", Optional.ofNullable(scopeType),
                "scopeId", Optional.ofNullable(scopeId),
                "limit", Optional.ofNullable(limit),
                "offset", Optional.ofNullable(offset)
        );
        ReportArchiveList responseObject = this.httpClient.get(url + "/reports/archives/", new HttpRequestConfig(queryParams), ReportArchiveList.class);
        return ReportArchiveList.to(responseObject);
    }`

Reply:
> What do you think if I add just i simple if condition in the controller?

Maybe we can extract this condition in some helper method within the `ReportsApi` class? e.g. `getReportArchivesPrefix(userId)`

Reply:
Done! Tell me what you think

Reply:
Thank you!

The new method seems to be too generic for this class - it's not clear that it's responsible for the report archive URLs. I would suggest something like this:

```java
getReportArchivesPath(Long userId, String path) {
  return userId != null ? String.format("%s/users/%d/%s", this.url, userId, path) : String.format("%s/%s", this.url, path);
}
```

I would also suggest mentioning in the method annotations that the `userId` parameter is only supposed to be used in crowdin.com: `@param userId    user identifier (for crowdin.com only)`

Reply:
It's not only for 'reportArchive' if you noticed, I've updated all the new endpoints where the 'userId' is required with my las commit. So new endpoints with 'userid' required uses this method.

Reply:
Btw, i've just noticed that I forgon 'userId' for 'checkReportArchiveExportStatus' :(

Reply:
So if I get it right, there is 6 new endpoints to add(6 for users & 6 for enterprizes). But the only difference is 'userId'.
So I wrote 6 endpots (1 endpoint) for both) with prefixed 'userId' if 'userId' is not 'null'

Reply:
I dont know why but sometime I get an encoding error in my IDE with files where ukranian language is used. I'm usig 'UTF-8' though. Do you have any ideas?

Reply:
Updated!

Reply:
> I dont know why but sometime I get an encoding error in my IDE with files where ukranian language is used. I'm usig 'UTF-8' though. Do you have any ideas?

Not sure, never noticed anything similar before